packages/safe-tools-client/app/components/truncated-blockchain-address/index.gts

```diff
@@ -0,0 +1,73 @@
+import Component from '@glimmer/component';
+import BoxelIconButton from '@cardstack/boxel/components/boxel/icon-button';
+import { on } from '@ember/modifier';
+import truncateMiddle from '@cardstack/safe-tools-client/helpers/truncate-middle';
+import cssVar from '@cardstack/boxel/helpers/css-var';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { later } from '@ember/runloop';
+import copyToClipboard from '@cardstack/boxel/helpers/copy-to-clipboard';
+import didInsert from '@ember/render-modifiers/modifiers/did-insert';
+
+import './index.css';
+
+interface Signature {
+  Element: HTMLDivElement;
+  Args: {
+    address: string;
+    isCopyable?: boolean | undefined;
+    copyIconColor?: string;
+  }
+}
+
+export default class TruncatedBlockchainAddress extends Component<Signature> {
+  @tracked isShowingCopiedConfirmation = 0;
+
+  @action registerListener(element: Element) {
+    element.addEventListener('click', (e) => {
+      e.stopPropagation();
+      return false;
+    });
+  }
+
+  @action flashCopiedConfirmation() {
+    this.isShowingCopiedConfirmation = 1;
+    later(() => {
+      this.isShowingCopiedConfirmation = 0;
+    } , 1000)
+  }
+
+  <template>
+    {{#if @isCopyable}}
+      <div class='truncated-blockchain-address' {{didInsert this.registerListener}}>
```

Comment:
This is a bit misleading - in both cases of `@isCopyable` the address will be truncated, but `truncated-blockchain-address` class is only used in the first case.